x-pack/plugins/observability/public/toggle_overview_link_in_nav.tsx

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreStart } from 'kibana/public';
+
+export function toggleOverviewLinkInNav(core: CoreStart) {
+  const { apm, logs, metrics, uptime } = core.application.capabilities.navLinks;
+  const someVisible = Object.values({ apm, logs, metrics, uptime }).some((visible) => visible);
+  if (!someVisible) {
+    core.chrome.navLinks.update('observability-overview', { hidden: true });
+  }
```

Comment:
Quite late, but note that this is a deprecated API:

https://github.com/elastic/kibana/blob/bf04235dae35452061cc7ea3d86d96c19a58206c/src/core/public/chrome/nav_links/nav_links_service.ts#L76-L82